### PR TITLE
start FlowCharts, import deps and add options

### PR DESCRIPTION
**Pull request by mshd**

### CODE REVIEW

**Comment by mshd** [comment]
have a look at the packages which I found:
- https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mermaid
- https://github.com/imjuni/erdia
- https://github.com/nurdiansyah/sveltekit-markdown/blob/2fffe9287943fced8516f33d4c5b19017f22a2af/src/lib/plugins/mermaid/index.ts
- https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mermaid
- https://github.com/stechy1/ast-to-mmd
- https://github.com/stechy1/ast-to-mmd/blob/ce20811c463981a2a509cc57ea7b2600c990252d/packages/ast-to-mmd/src/app/graph-blocks/renderer/shape-renderer.ts

**Comment by ogroppo** [comment]
have a look at the packages which I found:

https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mermaid - **OK**
https://github.com/imjuni/erdia - **RUBBISH**
https://github.com/nurdiansyah/sveltekit-markdown/blob/2fffe9287943fced8516f33d4c5b19017f22a2af/src/lib/plugins/mermaid/index.ts - **Not sure what they are trying to do here**
https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mermaid
https://github.com/stechy1/ast-to-mmd - **Smells interesting but unusable and no docs**
https://github.com/stechy1/ast-to-mmd/blob/ce20811c463981a2a509cc57ea7b2600c990252d/packages/ast-to-mmd/src/app/graph-blocks/renderer/shape-renderer.ts - **yep that's something we would need**

File: package.json [hunk 1]
```diff
@@ -5,9 +5,12 @@
     "test": "echo \"Error: no test specified\" && exit 1",
     "t": "ts-node"
   },
-  "dependencies": {},
+  "dependencies": {
+    "mermaid": "^8.14.0"
```
Comment (ogroppo) [inline package.json]: it's all dev packages actually as in prod mode this stuff is not used to generate docs, as of now :)

File: examples/myGraph.ts [hunk 2]
```diff
@@ -16,4 +18,12 @@ Post1.link(Comment1);
 Comment1.link(Comment1Like1);
 Comment1.link(Comment1Like2);
 
-new Graph([LonelyUser, User]).toHtml("myUserGraph", { showVerb: true });
+new Graph([LonelyUser, User]).toHtml(
+  "myUserGraph",
+  { showVerb: true },
+  {
+    fontFamily: "Courier New",
```
Comment (ogroppo) [inline examples/myGraph.ts]: nice, maybe the 2 config arguments could become one, hopefully the `mermaid.initialize(MMD_OPTIONS);` doesn't complain of the extraneous ones

File: src/lib/toMmdFlowChart.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Entity, Rel, Verbs } from "../classes/Entity";
+interface Options {
+  orientation: "LR" | "TB" | "BT" | "RL";
+}
+export default function toMMdFlowChart(root: Entity, options: Options) {
```
Comment (ogroppo) [inline src/lib/toMmdFlowChart.ts]: is `MMd` intentional? are you trying to make a function instead of the class?

Comment (mshd) [reply src/lib/toMmdFlowChart.ts]: I thought MMd stands for Mermaid?

File: package.json (hunk 1 above)
Comment (mshd) [reply package.json]: ok

File: src/lib/toMmdFlowChart.ts (hunk 3 above)
Comment (ogroppo) [reply src/lib/toMmdFlowChart.ts]: yes sure but I guess it would be MMD or Mmd, maybe the latter is less shouted

**Pull request merged**